File: Check Completeness of a Binary Tree.cpp
class Solution {
public:
    bool isCompleteTree(TreeNode* root) {
        queue<TreeNode*> q;
        q.push(root);
        bool flag=false;
        while(!q.empty())
        {
            TreeNode* t  = q.front();
            q.pop();
            if(t==NULL)
            {
                flag=true;
            }
            else
            {
                if(flag)
                {
                    return false;
                }
                q.push(t->left);
                q.push(t->right);
            }
        }
        return true;
    }
};

File: Check Completeness of a Binary Tree.java
class Solution {
    public boolean isCompleteTree(TreeNode root) {
        if (root == null) {
            return true;
        }
        
        Queue<TreeNode> queue = new LinkedList<>();
        queue.add(root);
        boolean seenNull = false;
        
        while (!queue.isEmpty()) {
            TreeNode node = queue.poll();
            
            if (node == null) {
                seenNull = true;
                continue;
            }
            
            if (seenNull) {
                return false;
            }
            
            queue.add(node.left);
            queue.add(node.right);
        }
        
        return true;
    }
}
